### PR TITLE
fixes for pipeline flow/dependencies

### PR DESCRIPTION
**Pull request by dsevost**
hello, dear okd-team,
please review the pull-req
1. `ovn-kubernetes` and `ovn-kubernetes-base` must be built via `Dockerfile` and `Dockerfile.base`, except for 4.13 branch ~(i see there is a new kustomization rule for 4.13 https://github.com/okd-project/okd-payload-pipeline/blob/a1c13616ef49d867bd48adcd3a890841a932e18d/buildconfigs/kustomization.yaml#L8 seems Dockerfile.*rhel9 migth be applied here)~
2. `ovn-kubernetes` depends on `ovn-kubernetes-base` and must be build on the next step of the pipeline
3. network-tools still refers to `registry.ci.openshift.org/ocp/4.12:ovn-kubernetes` as FROM

### CODE REVIEW

File: buildconfigs/20-ovn-kubernetes-base.yaml [hunk 1]
```diff
@@ -14,7 +14,8 @@ spec:
     type: Docker
     dockerStrategy:
       imageOptimizationPolicy: SkipLayers
-      dockerfilePath: Dockerfile.base.rhel9
+      # Dockerfile.rhel9 is suitable for 4.13 only
```
Comment (aleskandro) [inline buildconfigs/20-ovn-kubernetes-base.yaml:17]: 4.13 is the current stable and we should start rolling out changes to onboard 4.14 builds: this change leads again to a question we were discussing some time ago.

What versions should we support in this repo? 

@vrutkovs should we branch and create a release-4.12, release-4.13 and keep the master branch for 4.14?

File: variants/fcos/okd-rpms/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM release:artifacts as artifacts
```
Comment (aleskandro) [inline variants/fcos/okd-rpms/Dockerfile:1]: we want to avoid adding new Dockerfiles here.

File: buildconfigs/20-ovn-kubernetes-base.yaml (hunk 1 above)
Comment (vrutkovs) [reply buildconfigs/20-ovn-kubernetes-base.yaml:17]: Yeah, lets create release-4.13 branch at least. Not sure if we have capacity to actively support 4.12 builds tbh, but worth having for historical purposes at least

File: variants/fcos/okd-rpms/Dockerfile (hunk 2 above)
Comment (vrutkovs) [reply variants/fcos/okd-rpms/Dockerfile:1]: Right, we don't need those. `okd-rpms` is useful for Assisted Installer but not usable for SCOS installs

File: variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: machineconfiguration.openshift.io/v1
```
Comment (vrutkovs) [inline variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml:1]: Ditto, we should build SCOS with these options enabled (not sure if intel_pstate is necessary anymore)

File: buildconfigs/20-ovn-kubernetes-base.yaml (hunk 1 above)
Comment (dsevost) [reply buildconfigs/20-ovn-kubernetes-base.yaml:17]: branching is suitable here

File: variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml (hunk 3 above)
Comment (aleskandro) [reply variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml:1]: @vrutkovs we don't build scos here yet as we consume it directly from the artifacts produced by the okd-project/okd-scos pipeline.

Ideally, we should reach a target like:

- okd-project/okd-scos produces a base scos image (w/o okd-specific content)
- we build the layered okd-specific scos as we do for fcos (leveraging a third repo similar to openshift/okd-machine-os)
- 
right?

File: variants/fcos/okd-rpms/Dockerfile (hunk 2 above)
Comment (dsevost) [reply variants/fcos/okd-rpms/Dockerfile:1]: just for my understading: will release:artifacts and release:installer artifacts be converged further?

**Comment by dsevost** [comment]
included into https://github.com/okd-project/okd-payload-pipeline/pull/39#issue-1746710513

**Pull request closed without merging**